src/vanilla/store.ts

```diff
@@ -247,7 +247,11 @@ const flushBatch = (batch: Batch) => {
 
 // internal & unstable type
 type StoreArgs = readonly [
-  getAtomState: <Value>(atom: Atom<Value>) => AtomState<Value>,
+  getAtomState: <Value>(atom: Atom<Value>) => AtomState<Value> | undefined,
+  setAtomState: <Value>(
+    atom: Atom<Value>,
+    atomState: AtomState<Value>,
+  ) => AtomState<Value>,
```

Comment:
nice 😎

Reply:
though, returning a (possibly modified) atomState is not very intuitive.

Reply:
hmm, maybe this could help, though its a little pedantic
```ts
setAtomState: <Value, AS extends AtomState<Value>>(
    atom: Atom<Value>,
    atomState: AS,
  ) => AS,
```

Reply:
Good idea. The arg must be original one.